src/main/java/seedu/exercise/storage/JsonAdaptedCustomProperty.java

```diff
@@ -0,0 +1,114 @@
+     */
+    private Prefix toModelShortName() throws IllegalValueException {
+        if (shortName == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                Prefix.class.getSimpleName()));
+        }
+        if (!CustomProperty.isValidShortName(shortName)) {
+            throw new IllegalValueException(CustomProperty.SHORT_NAME_CONSTRAINTS);
+        }
+        return new Prefix(shortName + "/");
+    }
+
+    /**
+     * Converts the full name of a Jackson-friendly adapted custom property object into its corresponding
+     * model's name.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted custom property
+     */
+    private String toModelFullName() throws IllegalValueException {
+        String fullNameField = "Full Name";
+        if (fullName == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, fullNameField));
+        }
+        if (!CustomProperty.isValidFullName(fullName)) {
+            throw new IllegalValueException(CustomProperty.FULL_NAME_CONSTRAINTS);
+        }
+        return fullName.trim();
+    }
+
+    /**
+     * Converts the parameter type of a Jackson-friendly adapted custom property object into its corresponding
+     * model's {@code ParameterType} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted custom property
+     */
+    private ParameterType toModelParameterType() throws IllegalValueException {
+        if (parameterType == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                ParameterType.class.getSimpleName()));
+        }
+
+        if (!ParameterType.isValidParameterType(parameterType)) {
+            throw new IllegalValueException(ParameterType.PARAMETER_CONSTRAINTS);
+        }
+
+        if (parameterType.equals(ParameterType.NUMBER.getParameterName())) {
+            return ParameterType.NUMBER;
+        } else if (parameterType.equals(ParameterType.TEXT.getParameterName())) {
+            return ParameterType.TEXT;
+        } else {
+            return ParameterType.DATE;
+        }
+    }
+
+    /**
+     * Converts the short name of a Jackson-friendly adapted custom property object into its corresponding
+     * model's {@code Prefix} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted custom property
+     */
+    public CustomProperty toModelType() throws IllegalValueException {
+        final Prefix modelShortName = toModelShortName();
+        final String modelFullName = toModelFullName();
+        final ParameterType modelParameterType = toModelParameterType();
+        return new CustomProperty(modelShortName, modelFullName, modelParameterType);
+    }
```

Comment:
it is not specified anywhere but i think that public methods should generally come before private methods. Not really a big deal but makes it easier to read

Reply:
Have made the changes 👍